yara/yara/core_yara.c

```diff
@@ -95,12 +96,17 @@ static int callback (YR_SCAN_CONTEXT* context, int message, void *msg_data, void
 			YR_MATCH* match;
 			yr_string_matches_foreach(context, string, match)
 			{
+				// Find virtual address if possible
+				RBinSection *s = r_bin_get_section_at (core->bin->cur->o, match->base + match->offset, false);
+				if (s != NULL) {
+					offset =  s->vaddr - s->paddr;
+				}
 				const char *flag = sdb_fmt ("%s%d_%s_%d", "yara", flagidx, rule->identifier, ruleidx);
 				if (print_strings) {
-					r_cons_printf("0x%08" PRIx64 ": %s : ", match->base + match->offset, flag);
+					r_cons_printf("0x%08" PRIx64 ": %s : ", match->base + match->offset + offset, flag);
```

Comment:
add a space before ( and use PFMT64x which is the macro used in all the r2 codebase for portable ut64  hexa